### ISSUE TITLE
Add a shellcheck rule for use of echo

### ISSUE THREAD

**Issue by LLyaudet** [koalaman/shellcheck#3419]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows a potential problem:

```sh
#!/bin/sh
# shellcheck enable=all
foo="abcd"
echo "${foo}"
echo "success XD"
```

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted to see:
Warning on line xxx : echo "${foo}"
This is bad, variable could have dash prefix.
Warning on line xxx : echo "success XD"
This is bad, POSIX told you that echo was deprecated for incompatible implementations between POSIX systems.

For more than 2 decades now,
POSIX says that newer applications should use echo.
https://pubs.opengroup.org/onlinepubs/000095399/
Either it was intended as a joke or a canary trap (who got that information?),
or there should be an optional warning in shellcheck to enforce this in new application ;).
